tools/CAT/src/components/EditDefaultInput/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { Input } from "@chakra-ui/react";
+import { useState, useEffect } from "react";
+import SaveButton from "../SaveButton";
+
+interface EditDefaultInputProps {
+	className?: string;
+	value: string;
+	placeholder?: string;
+	onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export default function EditDefaultInput({
+	onChange,
+	value,
+	onClick,
+	placeholder,
+	className,
+}: EditDefaultInputProps) {
+	const [editValue, setEditValue] = useState<string>(value);
+	const [isShowButton, setIsShowButton] = useState<boolean>(false);
+	const [initialValue, setInitialValue] = useState<string>(value);
+
+	useEffect(() => {
+		if (editValue === initialValue) {
+			setIsShowButton(false);
+		} else {
+			setIsShowButton(true);
+		}
+	}, [editValue, initialValue]);
+
+	const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setEditValue(event.target.value);
+		onChange(event);
+	};
+
+	return (
+		<div className={`flex w-fit max-w-fit h-fit ${className}`}>
```

Comment:
What will be if we dont pass className to component? 
I think will be this line

`flex w-fit max-w-fit h-fit undefined`

but its only my opinion